testinfra/backend/ansible.py

```diff
@@ -87,4 +87,15 @@ def get_variables(self) -> dict[str, Any]:
     @classmethod
     def get_hosts(cls, host: str, **kwargs: Any) -> list[str]:
         inventory = kwargs.get("ansible_inventory")
-        return AnsibleRunner.get_runner(inventory).get_hosts(host or "all")
+        hosts = AnsibleRunner.get_runner(inventory).get_hosts(host or "all")
+        limit = kwargs.get("ansible_limit")
+        if limit:
+            # Filter hosts based on the limit expression
+            from ansible.parsing.dataloader import DataLoader
+            from ansible.inventory.manager import InventoryManager
```

Comment:
I think it will be clearer to bring the imports to the top of the file

Reply:
@osk-8 done :)